docs/init.md

```diff
@@ -0,0 +1,201 @@
+$this->add_cart_handler(new WCS_Cart_Initial_Payment());
+
+// Some classes are initialized later in the request lifecycle
+add_action('init', array('WC_Subscriptions_Synchroniser', 'init'));
+add_action('after_setup_theme', array('WC_Subscriptions_Upgrader', 'init'), 11);
+add_action('init', array('WC_PayPal_Standard_Subscriptions', 'init'), 11);
+```
+
+### Hooks Registration
+
+The `init_hooks()` method sets up the following key hooks:
+
+1. Registration of custom order types and statuses
+2. Loading of translations
+3. Adding plugin action links
+4. Activation/deactivation procedures
+5. Action Scheduler batch size configuration
+6. Initialization of notification batch processor
+
+## Context-Specific Initialization
+
+### Admin Pages
+
+When WooCommerce Subscriptions Core is loaded in the WordPress admin:
+
+1. The core initialization flow runs first
+2. `WC_Subscriptions_Admin::init()` is called, which:
+   - Loads admin-specific scripts and styles
+   - Registers admin menus and settings
+   - Sets up meta boxes for subscription editing
+   - Configures admin notices
+
+3. Version-dependent admin classes are initialized via `init_version_dependant_classes()`:
+   ```php
+   new WCS_Admin_Post_Types();
+   new WCS_Admin_Meta_Boxes();
+   // ... other admin-specific components
+   ```
+
+4. Admin-specific hooks are registered for:
+   - Order management screens
+   - Product management screens
+   - Payment gateway configuration
+   - Reporting and analytics
+
+### Customer-Facing Pages
+
+For front-end pages visible to customers:
+
+1. The core initialization flow runs
+2. Front-end specific components are initialized:
+   - `WC_Subscriptions_Cart` - manages subscription products in the cart
+   - `WC_Subscriptions_Checkout` - handles checkout flow for subscriptions
+   - `WC_Subscriptions_Frontend_Scripts` - loads necessary JS/CSS
+
+3. No admin-specific components are loaded, which keeps front-end pages lean
+4. `WCS_Template_Loader` manages template overrides and customizations
+
+### Action Scheduler Events
+
+When an Action Scheduler event runs for subscriptions:
+
+1. WordPress core loads and the request is identified as a `cron` request (Action Scheduler uses WP Cron)
+2. The core initialization flow runs, but admin UI components are not loaded
+3. `WCS_Action_Scheduler` (initialized during core initialization) processes the scheduled event:
+   - Retrieves the correct subscription by ID from the action arguments
+   - Performs the scheduled action (payment processing, trial end, etc.)
+   - Updates subscription dates and statuses as appropriate
+
+4. The subscription status may change based on the event, which then triggers:
+   - Email notifications
+   - Payment processing
+   - Status changes and associated actions
+
+### REST API Requests
+
+For REST API requests related to subscriptions:
+
+1. WordPress identifies the request as a REST API request via `WC()->is_rest_api_request()`
```

Comment:
This isn't quite right. *WordPress* definitely does not rely on WooCommerce to identify the request as being related to the REST API 🙃

Reply:
Weird, but there is a helper function `wcs_is_rest_api_request` which does this check if available. Anyway, that's not enough to be confident, so I replaced with reference to helper function only.